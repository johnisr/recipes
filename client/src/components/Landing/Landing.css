.carousel {
  text-align: center;
}

.slider {
  border-radius: 5px;
}

.slider__container {
  position: relative;
}

.slider__title {
  margin-bottom: 0;
}
.slider__subtitle {
  margin-top: 0;
}

.slider__image {
  border-radius: 5px;
}

.slider__button--left {
  position: absolute;
  top: 50%;
  left: 0;
  transform: translateY(-50%);
}

.slider__button--right {
  position: absolute;
  top: 50%;
  right: 0;
  transform: translateY(-50%);
}

.slider__link {
  position: absolute;
  bottom: 0;
  right: 0;
  transform: translate(-50%, -50%);
}

/*  need 3 levels of specificity to beat semantic styling
    react-carousel active = disabled, semantic ui has disabled
    with opacity 0.45 !important
*/
.slider__container .slider__dot--active[role='button'],
.slider__container .slider__dot--active[role='button']:focus {
  background-color: white;
  opacity: 1 !important;
}

.slider__container .slider__dot[role='button'],
.slider__container .slider__dot[role='button']:focus {
  background-color: #e8e8e8;
  opacity: 0.5 !important;
}

.slider__dots {
  position: absolute;
  bottom: 2%;
  left: 50%;
  transform: translateX(-50%);
}
