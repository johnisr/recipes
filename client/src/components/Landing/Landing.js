import React from 'react';

const Landing = () => {
  return (
    <div>
      Landing!
      <div>Click Recipes to see a list of recipes</div>
      <div>Login to add recipes</div>
    </div>
  );
};

export default Landing;
