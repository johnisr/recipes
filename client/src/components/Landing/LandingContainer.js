import React from 'react';
import { CarouselProvider } from 'pure-react-carousel';
import { Loader } from 'semantic-ui-react';
import { connect } from 'react-redux';
import Landing from './Landing';
import seededRandom from '../../utils/seededRandom';

const LandingContainer = props => {
  const { recipes, seed } = props;
  if (recipes === null) {
    return <Loader active content="Loading" size="large" />;
  }

  let start = seed;
  const randomRecipes = recipes
    .filter(recipe => recipe.imageUrl && !!recipe.imageUrl[0])
    // eslint-disable-next-line no-plusplus
    .sort(() => 0.5 - seededRandom(start++));

  return (
    <CarouselProvider
      naturalSlideWidth={100}
      naturalSlideHeight={70}
      totalSlides={randomRecipes.length}
      interval={10000}
      isPlaying
    >
      <Landing recipes={randomRecipes} />
    </CarouselProvider>
  );
};

const mapStateToProps = ({ recipes, carousel }) => ({
  recipes,
  seed: carousel.seed,
});

export default connect(mapStateToProps)(LandingContainer);
