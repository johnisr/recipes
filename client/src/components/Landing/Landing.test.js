import React from 'react';
import { shallow } from 'enzyme';
import { Landing } from './Landing';

test('should render correctly', () => {
  const wrapper = shallow(<Landing recipes={[]} />);
  expect(wrapper).toMatchSnapshot();
});
