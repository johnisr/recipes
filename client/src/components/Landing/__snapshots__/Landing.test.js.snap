// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Landing!",
        <div>
          Click Recipes to see a list of recipes
</div>,
        <div>
          Login to add recipes
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Landing!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click Recipes to see a list of recipes",
        },
        "ref": null,
        "rendered": "Click Recipes to see a list of recipes",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login to add recipes",
        },
        "ref": null,
        "rendered": "Login to add recipes",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Landing!",
          <div>
            Click Recipes to see a list of recipes
</div>,
          <div>
            Login to add recipes
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Landing!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click Recipes to see a list of recipes",
          },
          "ref": null,
          "rendered": "Click Recipes to see a list of recipes",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login to add recipes",
          },
          "ref": null,
          "rendered": "Login to add recipes",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
